tests/disas/indirect-call-caching-exclude-0-index.wat

```diff
@@ -0,0 +1,102 @@
+;;! target = "x86_64"
+;;! flags = [ "-Wcache-call-indirects=y" ]
+
+(module
```

Comment:
Care to add a brief comment to this test file about what we are aiming to exercise here, so that future changes that happen to tweak this disassembly know what property to check is preserved in their changes?

Reply:
For sure, added explanations here and below giving the reasons for expecting/not expecting the opt, and the key bits of IR we're asserting.